frontend/pages/login.tsx

```diff
@@ -0,0 +1,108 @@
+import { GetServerSideProps } from "next";
+import { useRouter } from "next/router";
+
+import { useState } from "react";
+
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+
+import Navbar from "../components/Navbar/Narbar";
+import { useApiClient } from "../hooks/api";
+import styles from "../styles/Login.module.css";
+
+type LoginProps = {
+  host: string;
+};
+
+type LoginFormValues = {
+  email: string;
+  password: string;
+};
+
+const Login = (props: LoginProps) => {
+  const router = useRouter();
+
+  const { apiClient, setHost } = useApiClient();
+  setHost(props.host);
+
+  //error handling
+  const [loginError, setLoginError] = useState("");
```

Comment:
It's better practice to do this instead:
```suggestion
  const [loginError, setLoginError] = useState<string>();
```